src/data/polynomial/denominators.lean

```diff
@@ -0,0 +1,90 @@
+/-
+Copyright (c) 2020 Damiano Testa. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Damiano Testa
+-/
+import data.polynomial.erase_lead
+
+/-!
+# Denominators of evaluation of polynomials at ratios
+
+Let `i : R → K` be a homomorphism of semirings.  Assume that `K` is commutative.  If `a` and
+`b` are elements of `R` such that `i b ∈ K` is invertible, then for any polynomial
+`f ∈ polynomial R` the "mathematical" expression `b ^ f.nat_degree * f (a / b) ∈ K` is in
+the image of the homomorphism `i`.
+-/
+
+open polynomial finset
+
+/--  An induction lemma for polynomials. It takes a natural number `N` as a parameter, that is
+required to be at least as big as the `nat_degree` of the polynomial.  This is useful to prove
+results where you want to change each term in a polynomial to something else depending on the
+`nat_degree` of the polynomial itself and not on the specific `nat_degree` of each term. -/
+lemma induction_with_nat_degree_le {R : Type*} [semiring R] {P : polynomial R → Prop} (N : ℕ)
```

Comment:
As you suggested, please move this to the file on induction principles for polynomials.

Reply:
I tried moving it to the `induction` file.  I then have to import the `erase_lead` file, which causes a cyclic import.  Maybe I can put the induction lemma in the file `erase_lead` instead?

Reply:
yup, looks good

Reply:
Great!  Besides, I had pushed this change independently, anyway!